src/configuration/decoration.ts

```diff
@@ -24,4 +25,13 @@ export class Decoration {
   static readonly EasyMotion = vscode.window.createTextEditorDecorationType({
     backgroundColor: configuration.searchHighlightColor,
   });
+
+  static yankHighlight(vimState: VimState, range: vscode.Range[]) {
+    const decoration = vscode.window.createTextEditorDecorationType({
+      backgroundColor: configuration.yankHighlightColor,
+    });
+
+    vimState.editor.setDecorations(decoration, range);
+    setTimeout(() => decoration.dispose(), 200);
```

Comment:
Interesting. Is this dispose necessary? As per the documentation:

```
		/**
		 * Remove this decoration type and all decorations on all text editors using it.
		 */
```

Doesn't sound like that's what we want.

Reply:
Can we do the actual `setDecorations` outside of this? As it stands, this class is primarily creating the decorationType and it's the responsibility of the caller to actually perform the set.

Reply:
Regarding the call to `dispose`; preferably no. The problem occurs when using multiple cursors.

When each cursor performs a yank operation there are multiple affected ranges. The best case scenario would be to aggregate all these ranges and call `setDecorations` only once (with a static `YankHighlight` instance), but I could not find an applicable place to implement this logic.

Therefore I could only see two alternatives:
- Use a static `YankHighlight` decorator instance and call `setDecorations` for each cursor. The problem with this approach is that each cursor replaces the previous cursor's decoration (since the instance is reused), and therefore only highlights the last yank operation.
- Create a unique `YankHighlight` decorator instance for each cursor, disposing it (i.e removing the highlight) after a specific timeout has passed (the current approach).

But if there is a reasonable way to aggregate all affected ranges and apply the `setDecorations` logic once all cursors' yank operation has finished executing I'd gladly implement it.

Reply:
Ah, sorry didn't realize that we wanted the `yank` to disappear. I think the way it is at present is fine.